### PR TITLE
Small LFS additions, part 4

### PR DESCRIPTION
**Pull request by no92**

### CODE REVIEW

File: abis/linux/in.h [hunk 1]
```diff
@@ -44,6 +44,8 @@ struct ipv6_mreq {
 	unsigned        ipv6mr_interface;
 };
 
+extern const struct in6_addr in6addr_any;
```
Comment (ArsenArsen) [inline abis/linux/in.h:47]: this isn't defined afaict?

File: options/ansi/include/setjmp.h [hunk 2]
```diff
@@ -12,7 +12,7 @@ extern "C" {
 
 // [C11/7.13] Non-local jumps
 
-typedef struct {
+typedef struct __jmp_buf {
```
Comment (ArsenArsen) [inline options/ansi/include/setjmp.h:15]: what's this used for?

File: options/posix/generic/unistd-stubs.cpp [hunk 3]
```diff
@@ -683,9 +684,13 @@ unsigned long sysconf(int number) {
 		case _SC_PAGE_SIZE:
 			return mlibc::page_size;
 		case _SC_OPEN_MAX:
-			// TODO: actually return a proper value for _SC_OPEN_MAX
-			mlibc::infoLogger() << "\e[31mmlibc: sysconf(_SC_OPEN_MAX) returns arbitrary value 256\e[39m" << frg::endlog;
-			return 256;
+			struct rlimit ru;
+			if (getrlimit(RLIMIT_NOFILE, &ru) < 0) {
```
Comment (ArsenArsen) [inline options/posix/generic/unistd-stubs.cpp:688]: This function can't fail with anything but ENOSYS, might be worth asserting for that when logging

File: options/posix/generic/unistd-stubs.cpp [hunk 4]
```diff
@@ -683,9 +684,13 @@ unsigned long sysconf(int number) {
...
+				mlibc::infoLogger() << "\e[31mmlibc: sysconf(_SC_OPEN_MAX) returns arbitrary value 256\e[39m" << frg::endlog;
+				return 256;
+			} else {
+				return ru.rlim_cur;
```
Comment (ArsenArsen) [inline options/posix/generic/unistd-stubs.cpp:692]: if this is infinity, it should return -1 AFAIK

**Review by ArsenArsen — CHANGES_REQUESTED** [review]

File: options/ansi/include/setjmp.h (hunk 2 above)
Comment (Dennisbonke) [reply options/ansi/include/setjmp.h:15]: I honestly don't remember it but something needed it.

File: abis/linux/in.h (hunk 1 above)
Comment (Dennisbonke) [reply abis/linux/in.h:47]: Hmm, likely that this isn't needed at all as this line is already declared in `options/posix/include/netinet/in.h`. The actual definition is in `options/posix/generic/arpa-inet-stubs.cpp`. This should probably be dropped.

File: options/posix/generic/unistd-stubs.cpp (hunk 3 above)
Comment (no92) [reply options/posix/generic/unistd-stubs.cpp:688]: On linux, the man page states that a other errno values can be set, namely `EFAULT` and `EINVAL`.

Comment (ArsenArsen) [reply options/posix/generic/unistd-stubs.cpp:688]: yes, and neither is possible here, unless NOFILE is not implemented

**Review by Geertiebear — APPROVED** [review]
LGTM

**Pull request merged**